File: tests/fixtures/java/InjectionIssues.java
// Test fixtures for Java injection vulnerabilities
import java.sql.*;
import java.io.IOException;

public class InjectionIssues {

    // Should trigger JAVA_INJ001
    public void sqlInjection(Connection conn, String userId) throws SQLException {
        String query = "SELECT * FROM users WHERE id = " + userId;
        Statement stmt = conn.createStatement();
        ResultSet rs = stmt.executeQuery(query);
    }

    // Should trigger JAVA_INJ001
    public void sqlInjectionWithConcat(Connection conn, String username) throws SQLException {
        String query = "SELECT * FROM users WHERE username = '" + username + "'";
        Statement stmt = conn.createStatement();
        stmt.executeQuery(query);
    }

    // Should trigger JAVA_INJ002
    public void commandInjectionRuntime(String userInput) throws IOException {
        Runtime.getRuntime();
    }

    // Should trigger JAVA_INJ002
    public void commandInjectionProcessBuilder(String path) throws IOException {
        ProcessBuilder pb = new ProcessBuilder("cat", path);
    }
}

File: tests/fixtures/rust/injection_issues.rs
// Test fixtures for Rust injection vulnerabilities

fn test_sql_injection(user_id: &str) {
    // Should trigger RUST_INJ001
    let query = format!("SELECT * FROM users WHERE id = {}", user_id);
    println!("{}", query);

    let update = format!("UPDATE users SET name = '{}' WHERE id = 1", user_id);
    println!("{}", update);
}

fn test_string_formatting_sql() {
    let user_input = "malicious'; DROP TABLE users--";

    // Should trigger RUST_INJ001
    println!("SELECT * FROM products WHERE name = '{}'", user_input);
}
